### PR TITLE
Added `scipy.special.binom` functionality to CuPy

### PR DESCRIPTION
**Pull request by KyanCheung**
My first PR. I've added test cases, so hopefully this is good to go.

### CODE REVIEW

File: cupyx/scipy/special/_combinatorics.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        p1evl_definition +
+        polevl_definition +
+        lgam_sgn_definition +
+        lbeta_symp_definition +
+        lbeta_definition +
+        beta_definition +
+        binom_definition +
+        comb_definition 
+    ),
+    doc="""The number of combinations of N things taken k at a time.
```
Comment (emcastillo) [inline cupyx/scipy/special/_combinatorics.py:77]: Shouldn't this function being imported in `__init__` and have its tests too? 😃

**Review by emcastillo — CHANGES_REQUESTED** [review]
Looks neat!
Please either add tests for `comb` or lets remove it from this PR is you plan to work on it later :)

**Comment by KyanCheung** [comment]
Ah sorry, didn't mean to upload `comb` as well. I programmed it, but decided that I'd only upload `binom` first afterwards. 

After this PR finishes, I'll make a separate PR for `comb` and `perm` implementation.

**Review by emcastillo — APPROVED** [review]
Awesome work! thanks!

**Comment by emcastillo** [comment]
seems like CI failures are real

**Comment by KyanCheung** [comment]
Sorry all, I've been busy with other things and I couldn't fix the tests.

I've checked `test_binom` on my GPU and things are luckily working now on my end.

**Comment by KyanCheung** [comment]
Seems like the errors were caused by the error tolerance being too tight. That being said, there were 22 failures and I only had 6 tests. Something else going on?

**Comment by KyanCheung** [comment]
The failed tests all seem to be unrelated to the code I added?

**Comment by KyanCheung** [comment]
Also one thing I've noticed now - do I add `#NOQA` to the function import as well in `_init_`? What does it do?

File: cupyx/scipy/special/__init__.py [hunk 2]
```diff
@@ -83,6 +83,7 @@
 from cupyx.scipy.special._sph_harm import sph_harm  # NOQA
 
 # Other special functions
+from cupyx.scipy.special._binom import binom
```
Comment (emcastillo) [inline cupyx/scipy/special/__init__.py:86]:
```suggestion
from cupyx.scipy.special._binom import binom  # NOQA
```

**Comment by emcastillo** [comment]
> Also one thing I've noticed now - do I add #NOQA to the function import as well in _init_? What does it do?

It is used to avoid linter errors

File: cupyx/scipy/special/_binom.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    Returns
+    -------
+    scalar or ndarray
+        Value of binomial coefficient
+
+    .. seealso:: :func:`scipy.special.binom`
+
+    """,
+)
```
Comment (emcastillo) [inline cupyx/scipy/special/_binom.py:118]:
```suggestion
)

```

File: tests/cupyx_tests/scipy_tests/special_tests/test_binom.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        k = xp.linspace(.5-1e10, .5-9e10, 1000, dtype=dtype)
+        return scp.special.binom(n, k)
+    
+    @testing.for_float_dtypes()
+    @testing.numpy_cupy_allclose(atol=1e-5, rtol=1e-5, scipy_name='scp')
+    def test_nan_inf(self, xp, scp, dtype):
+        import scipy.special  # NOQA
+        a = xp.array([-numpy.inf, numpy.nan, numpy.inf, 0, -1,
+                      1e8, 5e7], dtype=dtype)
+        return scp.special.binom(a[:, xp.newaxis], a[xp.newaxis, :])
```
Comment (emcastillo) [inline tests/cupyx_tests/scipy_tests/special_tests/test_binom.py:56]:
```suggestion
        return scp.special.binom(a[:, xp.newaxis], a[xp.newaxis, :])

```

**Review by emcastillo — APPROVED** [review]

**Pull request merged**